### PR TITLE
Session API review changes

### PR DESCRIPTION
**Pull request by kichalla**
@Tratcher (This PR builds on top of the previous PR related to referencing Http.Features in Http.Abstractions)

Please note that I did not make the change related to removing the `UseInMemorySession` and `UseDistributedSession`.

### CODE REVIEW

**Comment by Tratcher** [comment]
Why not remove UseInMemorySession?

**Comment by kichalla** [comment]
MVC repo uses UseInMemorySession and I did not find it to be easier to use what the UseInMemorySession currently does

**Comment by Tratcher** [comment]
Take it up with @davidfowl then.

File: src/Microsoft.AspNet.Session/DistributedSession.cs [hunk 1]
```diff
@@ -124,13 +121,21 @@ public void Commit()
...
                 Serialize(stream);
-                _cache.Set(
+                await _cache.SetAsync(
                     _sessionId,
                     stream.ToArray(),
                     new DistributedCacheEntryOptions().SetSlidingExpiration(_idleTimeout));
             }
         }
 
+        private void Load()
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Session/DistributedSession.cs]: nit: put this next to LoadAsync.

**Comment by Tratcher** [comment]
So far so good.

File: test/Microsoft.AspNet.Session.Tests/project.json [hunk 2]
```diff
@@ -3,7 +3,8 @@
     "Microsoft.AspNet.Session": "1.0.0-*",
     "Microsoft.AspNet.TestHost": "1.0.0-*",
     "Microsoft.Framework.Logging.Testing": "1.0.0-*",
-    "xunit.runner.aspnet": "2.0.0-aspnet-*"
+    "xunit.runner.aspnet": "2.0.0-aspnet-*",
+	"Microsoft.Framework.Caching.Abstractions": "1.0.0-*"
```
Comment (Tratcher) [inline test/Microsoft.AspNet.Session.Tests/project.json]: nit: formatting

File: src/Microsoft.AspNet.Session/SessionMiddlewareExtensions.cs [hunk 3]
```diff
@@ -1,45 +1,16 @@
...
-using Microsoft.Framework.OptionsModel;
 
 namespace Microsoft.AspNet.Builder
 {
     public static class SessionMiddlewareExtensions
     {
-        public static IApplicationBuilder UseInMemorySession([NotNull] this IApplicationBuilder app, IMemoryCache cache = null, Action<SessionOptions> configure = null)
+        public static IApplicationBuilder UseSession([NotNull] this IApplicationBuilder app)
         {
-            return app.UseDistributedSession(new LocalCache(cache ?? new MemoryCache(new MemoryCacheOptions())), configure);
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Session/SessionMiddlewareExtensions.cs]: You also need to remove Session's direct dependency on the Caching.Memory package (and add it to tests instead).

Comment (Tratcher) [inline src/Microsoft.AspNet.Session/SessionMiddlewareExtensions.cs]: Are you still using anything out of Http.Extensions?

**Comment by Tratcher** [comment]
:shipit:  after cleaning up the dependencies.

**Pull request merged**